pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -604,6 +604,12 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, _ v1.PodStatus, podStat
 		podIP = podStatus.IP
 	}
 
+	rawHostIP, err := m.runtimeHelper.GetHostIP()
+	hostIP := rawHostIP.String()
```

Comment:
I think `hostIP` can be an internal field in the `Kubelet` struct and `GetHostIP` can set the field if blank. This would avoid having to call `GetNodeInfo` for every pod. I can make this change either in this PR or a follow up PR, let me know :)